cuqi/experimental/mcmc/_sampler.py

```diff
@@ -238,7 +206,28 @@ def warmup(self, Nb, tune_freq=0.1) -> 'SamplerNew':
         return self
     
     def get_state(self) -> dict:
-        """ Return the state of the sampler. """   
+        """ Return the state of the sampler. 
+
+        The state is used when checkpointing the sampler.
+
+        The state of the sampler is a dictionary with keys 'metadata' and 'state'.
+        The 'metadata' key contains information about the sampler type.
+        The 'state' key contains the state of the sampler.
+
+        For example, the state of a "MH" sampler could be:
+
+        state = {
+            'metadata': {
+                'sampler_type': 'MH'
+            },
+            'state': {
+                'current_point': np.array([...]),
+                'current_target_logd': -123.45,
```

Comment:
one thought here related to storing logd for all sampler. In some sense, logd is derived from `current_point`. If we want the sate to be minimal, maybe we restrict it to current point only without storing in logd values

Reply:
You are right. I moved the logd out of the base sampler class. It now currently recides in ProposalBasedSampler. It can be refactored further in the future.